### DISCUSSION TITLE
X96 Max+ Ultra support?

### DISCUSSION THREAD

**Discussion by OBSIDIAN404** [devmfc/debian-on-amlogic#54] [Q&A] [answered]
Tell me, do X96 Max+ Ultra images based on Amlogic S905X4 support them? I plan to take it, but I didn’t find it in the list of supported devices, only the processor is similar.

**Comment by devmfc** [comment]
It will not work at this moment, only gigabit Ethernet works om SC2,  X96 Max+ Ultra  is 100M. There is some clock missing. If you want to have a good, well supported by mainline Linux SoC : get an S905X3 based box. But be aware that most of them nowadays have unsupported WiFi+BT modules.

**Comment by VMax2** [comment]
Hi, i just had it working pretty good, just the network adapter is not working, i tried both the 2 configs (with or without gbit, i mean "box=x96x4" and "box=x96x4_gigabit") for the x905X4 cpu and nothing to do, i got the shell but no network, so, i had the network up plugging a usb3 network adapter, exactly the "AmazonBasics USB 3.0 to 10/100/1000 Gigabit Ethernet Adapter".
The LCD display works perfectly.

"lshw -C network" say the card (eth0) is DISABLED and under "dmesg" i found the following errors about eth0 (that is the built-in adapter):

meson8b-dwmac fdc00000.ethernet: Failed to reset the dma
meson8b-dwmac fdc00000.ethernet eth0: stmmac_hw_setup: DMA engine initialization failed
meson8b-dwmac fdc00000.ethernet eth0: __stmmac_open: Hw setup failed

meson8b-dwmac fdc00000.ethernet: Failed to reset the dma
meson8b-dwmac fdc00000.ethernet eth0: stmmac_hw_setup: DMA engine initialization failed
meson8b-dwmac fdc00000.ethernet eth0: __stmmac_open: Hw setup failed

About the rest, everything it's running like a charm, it would be really the top if you could fix the eth0 trouble so to avoid installig the usb network adapter, but, ehy, thank you so much for your great image in any case!!!

**Comment by VMax2** [comment]
Starting with config "box=s905x4_generic" give the following errors for eth0:


meson8b-dwmac fdc00000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
meson8b-dwmac fdc00000.ethernet eth0: validation of rmii with support 00,00000000,00000000,00006280 and advertisement 00,00000000,00000000,00006280 failed: -EINVAL
meson8b-dwmac fdc00000.ethernet eth0: __stmmac_open: Cannot attach to PHY (error: -22)

Also, the LCD display stay turned off

**Comment by VMax2** [comment, marked as answer]
I forgot to say that in all tests i used this image "Devmfc_Ubuntu-Noble_6.6.29-meson64_Minimal-24.04.29.img.xz" and the box is a X96Max+ Ultra.
So, to summarize my experience till now, the best configs are "box=x96x4" and "box=x96x4_gigabit" where for both of them everything works great (ok hdmi is not so good but usable for initial config), the LCD display too, except for the lan card (eth0) wich doesn't start.
So, to have a network, i installed this gigabit usb/lan card "AmazonBasics USB 3.0 to 10/100/1000 Gigabit Ethernet Adapter" wich it's recognized and work great.
That's all for now, hope before or late someone gonna fix the internal lan card, it would be great!! Greetings :)
